pkg/loki/loki.go

```diff
@@ -63,39 +63,41 @@ import (
 type Config struct {
 	Target       flagext.StringSliceCSV `yaml:"target,omitempty"`
 	AuthEnabled  bool                   `yaml:"auth_enabled,omitempty"`
-	HTTPPrefix   string                 `yaml:"http_prefix"`
+	HTTPPrefix   string                 `yaml:"http_prefix" doc:"hidden"'`
 	BallastBytes int                    `yaml:"ballast_bytes"`
 
 	// TODO(dannyk): Remove these config options before next release; they don't need to be configurable.
 	//				 These are only here to allow us to test the new functionality.
-	UseBufferedLogger bool `yaml:"use_buffered_logger"`
-	UseSyncLogger     bool `yaml:"use_sync_logger"`
+	UseBufferedLogger bool `yaml:"use_buffered_logger" doc:"hidden"`
+	UseSyncLogger     bool `yaml:"use_sync_logger" doc:"hidden"`
 
-	Common              common.Config               `yaml:"common,omitempty"`
 	Server              server.Config               `yaml:"server,omitempty"`
-	InternalServer      internalserver.Config       `yaml:"internal_server,omitempty"`
+	InternalServer      internalserver.Config       `yaml:"internal_server,omitempty" doc:"hidden"`
 	Distributor         distributor.Config          `yaml:"distributor,omitempty"`
 	Querier             querier.Config              `yaml:"querier,omitempty"`
-	CompactorHTTPClient compactor_client.HTTPConfig `yaml:"compactor_client,omitempty"`
-	CompactorGRPCClient compactor_client.GRPCConfig `yaml:"compactor_grpc_client,omitempty"`
-	IngesterClient      client.Config               `yaml:"ingester_client,omitempty"`
+	QueryScheduler      scheduler.Config            `yaml:"query_scheduler"`
+	Frontend            lokifrontend.Config         `yaml:"frontend,omitempty"`
+	QueryRange          queryrange.Config           `yaml:"query_range,omitempty"`
+	Ruler               ruler.Config                `yaml:"ruler,omitempty"`
+	IngesterClient      ingester_client.Config      `yaml:"ingester_client,omitempty"`
 	Ingester            ingester.Config             `yaml:"ingester,omitempty"`
-	StorageConfig       storage.Config              `yaml:"storage_config,omitempty"`
 	IndexGateway        indexgateway.Config         `yaml:"index_gateway"`
+	StorageConfig       storage.Config              `yaml:"storage_config,omitempty"`
 	ChunkStoreConfig    config.ChunkStoreConfig     `yaml:"chunk_store_config,omitempty"`
 	SchemaConfig        config.SchemaConfig         `yaml:"schema_config,omitempty"`
+	CompactorConfig     compactor.Config            `yaml:"compactor,omitempty"`
+	CompactorHTTPClient compactor_client.HTTPConfig `yaml:"compactor_client,omitempty" doc:"hidden"`
+	CompactorGRPCClient compactor_client.GRPCConfig `yaml:"compactor_grpc_client,omitempty" doc:"hidden"`
```

Comment:
Why are these clients hidden? At least the GRPC client should be visible for users to configure TLS settings.

Reply:
These flags were added recently and were still not yet documented in the `_index.md` file. Therefore, added the hidden flag for now. Nevertheless, will remove it in a follow-up PR in order to generate the documentation associated 👍